cmd/cofidectl/cmd/trustzone/trustzone.go

```diff
@@ -552,21 +553,42 @@ func validateOpts(opts addOpts) error {
 	if err != nil {
 		return err
 	}
-	if opts.kubernetesClusterOIDCIssuerURL != "" {
-		err = validateOIDCIssuerURL(opts.kubernetesClusterOIDCIssuerURL)
+	normalisedURL, err := validateOIDCIssuerURL(opts.kubernetesClusterOIDCIssuerURL)
+	if err != nil {
+		return fmt.Errorf("invalid --kubernetes-oidc-issuer: %w", err)
 	}
-	return err
+	opts.kubernetesClusterOIDCIssuerURL = normalisedURL
+
+	return nil
 }
 
-func validateOIDCIssuerURL(oidcIssuerURL string) error {
-	parsed, err := url.Parse(oidcIssuerURL)
+func validateOIDCIssuerURL(oidcIssuerURL string) (string, error) {
+	// It's an optional flag, so if it's empty, it's valid.
+	if oidcIssuerURL == "" {
+		return "", nil
+	}
+
+	u, err := url.ParseRequestURI(oidcIssuerURL)
 	if err != nil {
-		return err
+		return "", fmt.Errorf("invalid URL format: %w", err)
 	}
 
-	if !slices.Contains([]string{"http", "https"}, parsed.Scheme) {
-		return fmt.Errorf("unsupported scheme: %s", parsed.Scheme)
+	if u.Scheme != "https" {
+		return "", fmt.Errorf("URL scheme must be 'https', but got '%s'", u.Scheme)
 	}
 
-	return nil
+	if u.Host == "" {
+		return "", fmt.Errorf("URL must include a host")
+	}
+
+	if u.RawQuery != "" {
+		return "", fmt.Errorf("URL must not have a query component")
+	}
+
+	if u.Fragment != "" {
+		return "", fmt.Errorf("URL must not have a fragment component")
+	}
+
+	// Normalize by removing any trailing slash.
+	return strings.TrimRight(oidcIssuerURL, "/"), nil
```

Comment:
Better normalize the result of parsing. Also, why is the trailing slash necessarily wrong?

Reply:
If it's stored with the trailing slash, it 404s at the PSAT validation stage (ie `//`).